docs/DeveloperGuide.md

```diff
@@ -391,42 +395,51 @@ prevent entries retaining within pairingList after it has been deleted from clie
+has been added or not.
   ```
-        case COMMAND_LIST:
-            ArrayList<String> listCommandTypeAndFlags = getListCommandType(commandDetail);
-            boolean isListProperty = listCommandTypeAndFlags.get(0).trim().equals(PROPERTY_FLAG);
-            boolean isListClient = listCommandTypeAndFlags.get(0).equals(CLIENT_FLAG);
-            boolean isListEverything = listCommandTypeAndFlags.get(0).equals(EVERYTHING_FLAG);
-            if (isListProperty) {
-                return new ParseListProperty(listCommandTypeAndFlags.get(1));
-            } else if (isListClient) {
-                return new ParseListClient(listCommandTypeAndFlags.get(1));
-            } else if (isListEverything && listCommandTypeAndFlags.get(1).isEmpty()) {
-                return new ParseListEverything();
-            } else {
-                throw new UndefinedSubCommandTypeException(MESSAGE_INCORRECT_LIST_DETAILS);
-            }
+      private Parser parseListCommand(String commandDetail) throws UndefinedSubCommandTypeException {
+        ArrayList<String> listCommandTypeAndFlags = getListCommandType(commandDetail);
+        boolean isListProperty = listCommandTypeAndFlags.get(SUB_COMMAND_INDEX).trim().equals(PROPERTY_FLAG);
+        boolean isListClient = listCommandTypeAndFlags.get(SUB_COMMAND_INDEX).equals(CLIENT_FLAG);
+        boolean isListEverything = listCommandTypeAndFlags.get(SUB_COMMAND_INDEX).equals(EVERYTHING_FLAG);
+        boolean isListPairs = listCommandTypeAndFlags.get(SUB_COMMAND_INDEX).equals(PAIR_FLAG);
+        if (isListProperty) {
+            return new ParseListProperty(listCommandTypeAndFlags.get(COMMAND_FLAG_INDEX));
+        } else if (isListClient) {
+            return new ParseListClient(listCommandTypeAndFlags.get(COMMAND_FLAG_INDEX));
+        } else if (isListEverything && listCommandTypeAndFlags.get(COMMAND_FLAG_INDEX).isEmpty()) {
+            return new ParseListEverything();
+        } else if (isListPairs) {
+            return new ParseListPair(listCommandTypeAndFlags.get(COMMAND_FLAG_INDEX));
+        } else {
+            throw new UndefinedSubCommandTypeException(MESSAGE_INCORRECT_LIST_DETAILS);
+        }
+    }
   ```
 This block of code is part of ParseManager. It determines the type of list operation(-client, 
--property or -everything) and returns the corresponding object.  
+-property, -pair, or -everything) and returns the corresponding object.  
 Both ParseListClient and ParseListObject then determine if tags are present, and if they are valid, 
 throwing exceptions if any errors are encountered. They then return the corresponding Command type necessary.
 
-* There are five different classes which handle each of the features described above. Each class inherits from the
-   abstract Command class, and reads information present in either the PropertyList or ClientList objects respectively.
+* There are seven different classes which handle each of the features described above. Each class inherits from the
```

Comment:
For consistency, maybe when you put "7", your subsequent text should also be "7". Would be good if you standardise with one of them.